Create ratio system for simulation vs render distance

GotoFinal opened the pull request:
Current version of this plugin for 1.18 rather defeats the purpose of 1.18 update.  
Both simulation and view distance should be used separetly, this can be done in many way, i choosen the simplest one I can easly implement quickly for my friends server.  

In my version the performance calculation works only based on simulation performance, and view is just calculated using a ratio property. Additionally the view can be set to a static number and then simulation will still scale according to load. From quick testing simulation affect the performance much more, so thats simplest logical choice here.

More advanced modes could be added in future, like manually selecting ratio per level:
```
simulation:view
3:3
4:6
6:12
16:32
32:32
```
And mod could then generate in-between steps by this values based on ratio between 2 other steps, like: `3:3, 3:5, 4:6, 5:9, 6:12, 7:14, 8:16 ... 16:32, 17:32...` based on simple math to calculate steps between.  
Or do some stat work and figure out best way to scale them both separetly while keeping the server both playable and fun.

Some other options would be to also analyse ram usage for view distance, or network/amount of players itself.

But for now I believe this is still much better option than either not scaling simulation at all - and the mod because quite useless, or scaling it in the same way - and the mod just removed amazing 1.18 performance improvments of separating these 2 things.

I made these changes for my own server, just decided to share my approach, feel free to merge, create something own based on it, or just ignore.

henkelmax commented:
Thank you for your contribution.
Did you test this under "real life" conditions?

GotoFinal commented:
@henkelmax on my own friend closed server, it just vanilla-compatible 1.18 server, we have up to 20 players online, currently tested up to 17, so far seems to works nice, i changed only ratio to `2.5` as the performance impact is really low, so we can keep the simulation distance at like 8-12 but view distance at 20-30 when there is more players online. (well also changed min/max mstp to 35-45).

The only observation so far is that during bigger load the ratio is not perfect and it should be more logarithmic in some way to make it better. so during laggy time the simulation distance would be equals to view distance at min values, but when simulation can go back to like 10, then view distance often can go higher too to that 2x of current value. Kinda like i expected when i proposed 
```
simulation:view
3:3
4:6
6:12
16:32
32:32
```
solution.

Maybe the default for max simulation could be also changed to somehing lower, no one really cares if stuff simulates 32 chunks away, but being able to see 32 chunks away makes a big difference, im considering limiting my simulation to 10 on server - sadly with current solution and ratio of `2.0` it means that render distance wont go above `20`. So there is still much to improve in current solution.

henkelmax commented:
Just tested your changes and it is awesome!
Really good job!
